### ISSUE TITLE
[Error] zero counts of overdispersed genes 

### ISSUE THREAD

**Issue by ChoiJi-Hye** [dylkot/cNMF#83]
Hi, dylkot.
Thank you for developing the convenient tool.

I encountered the following error while trying to use your tool with an R script. It says to "Filter those cells and re-run or adjust the number of overdispersed genes." 
<img width="939" alt="image" src="https://github.com/dylkot/cNMF/assets/43129945/500e3016-ddc1-4eea-bc2f-8fdf483d26c0">

However, I want to remove cells. The error message only mentions 5 cells, making it difficult to remove the remaining cells. Is there a way to remove them?

Additionally, I'm curious if increasing the number of high variable genes might be a better approach than removing cells. Which method would you recommend?

Thank you for your help!
Ji-Hye Choi

**Comment by dylkot** [comment]
Hi Ji-Hye,

Are you filtering cells with low counts (e.g. less than 200 or 500)? My suspicion is that those cells causing the problem may have overall low cell counts and you can avoid the issue by filtering cells with low counts first. And are you using the default of 2000 overdispersed genes? You could maybe try 3000 but I wouldn't go much higher than that because eventually it introduces a lot of noise when you normalize genes to uniform variance.

I'll consider printing a full list of the offending cells in the future.

**Comment by ChoiJi-Hye** [comment]
Yes, I proceeded with the analysis using only the final remaining cells after removing those with fewer than 200 counts, and completing all additional processing and annotation steps. 
I utilized the default of 2000 variable genes, as suggested, cells with counts between 200 and 500 might have been problematic.

Even when attempting with 3000 genes, I encountered the same error with 12 cells. 
Fortunately, after removing only the 4 cells detected in the message from the results of the analysis performed with 2000 genes and rerunning it, it proceeded smoothly.

Thank you for the prompt response and assistance!
